### PR TITLE
Add collection conversion

### PR DESCRIPTION
**Pull request by SpencerC**
Contains #9
Contains #11

### CODE REVIEW

File: protoconvert.go [hunk 1]
```diff
@@ -123,20 +184,107 @@ func protoToValue(value *pb.Value) (any, error) {
...
 	}
 	return nil, fmt.Errorf("unsupported value type: %T, value: %+v", value.GetInner(), value.GetInner())
 }
+
+func protosToMap(values []*pb.Value, spec *pb.TypeSpec_Map) (any, error) {
+	l := len(values)
+	kts := spec.GetKey()
+	vts := spec.GetValue()
+
+	k0, err := protoToValue(values[0], kts)
```
Comment (mpenick) [inline protoconvert.go:219]: I think this might not handle empty maps correctly. It would panic?

**Review by mpenick — COMMENTED** [review]
It looks like in gocql you can also represent a `set` (and `list`) with `map[key]struct{}`. I think that might be a good idea for handling the `set` type in a way that golang devs expect: https://github.com/gocql/gocql/blob/5cf3afd187a9244635fc2ca7a7419465a4bffd06/marshal.go#L70

File: protoconvert.go (hunk 1 above)
Comment (SpencerC) [reply protoconvert.go:219]: Fixed.

**Comment by SpencerC** [comment]
> I think that might be a good idea for handling the `set` type in a way that golang devs expect

I'll add this if you guys want. I hesitate for three reasons:

1. Since there is currently no official way to represent a set, and the protocol buffers make requests and respond with slices, I don't think it's that unreasonable to require clients to decide the optimal way to convert from their own set representation to slices for themselves.
2. ILT has an open set proposal, so set support should be on the horizon: https://github.com/golang/go/discussions/47331. Note that this proposal suggests an `Add` method that takes a variadic (slice) argument, so making slices the de-facto set type for now means that clients can probably start using the official set type as soon as its released. Likewise, many community set libraries also initialize sets with slices.
3. IMHO, `map[T]struct{}` does not produce very readable code, and will confuse more junior developers.

**Comment by mpenick** [comment]
> IMHO, map[T]struct{} does not produce very readable code, and will confuse more junior developers.

I agree that it's not readable, but this is the way to represent a set in golang today w/o generics and https://github.com/golang/go/discussions/47331.

**Review by mpenick — APPROVED** [review]
Maybe add an issue for `map[key]struct{}` (it can be discussed in another ticket and doesn't need to block this PR). Otherwise, :+1:

**Pull request merged**